pkg/storage/pluginstore/grpc.go

```diff
@@ -358,10 +392,17 @@ func NewStruct(src map[string]interface{}) *structpb.Struct {
 }
 
 // AsMap converts a protobuf struct into its original representation, bson.M.
-func AsMap(src *structpb.Struct) bson.M {
+func AsMap(src *structpb.Struct, c ...converter) bson.M {
 	dest := src.AsMap()
+	converts := []converter{ConvertFloatToInt}
+	if c != nil {
+		converts = append(converts, c...)
+	}
```

Comment:
If you always want to run ConvertFloatToInt and then append extras that are passed, I think that you can simplify this a bit
```suggestion
	c = append(c, ConvertFloatToInt)
```

Then just use c instead of converts (or maybe rename c to converts since it's a better (more descriptive) parameter name.

Reply:
I simplified it to always run ConverFloatToInt before running passed in converters